api/v1alpha2/azuremachine_types.go

```diff
@@ -37,7 +38,9 @@ type AzureMachineSpec struct {
 	VMSize           string           `json:"vmSize"`
 	AvailabilityZone AvailabilityZone `json:"availabilityZone,omitempty"`
 
-	Image *Image `json:"image,omitempty"`
+	// Image is used to provide a reference to the image to use during VM creation.
+	// +optional
+	Image corev1.ObjectReference `json:"image,omitempty"`
```

Comment:
I think RuntimeExtension is deprecated, but I like that it doesn't require a CRD and allows you to embed the fields in the Machine manifest instead of a ref to another object.

Reply:
Thanks for the feeback. There's been another discussion on this attached to the issue (#293), essentially you end up with separate kinds, something like this:

```yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
kind: AzureImageMarketplace (or AzureMarketplaceImage)
metadata:
  name: node_image
spec:
  publisher: "Debian"
  offer: "debian-10"
  sku: "10"
  version: "latest"
---
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
kind: AzureMachineTemplate
metadata:
  name: ${CLUSTER_NAME}-md-0
spec:
  template:
    spec:
      location: ${AZURE_LOCATION}
      vmSize: ${NODE_MACHINE_TYPE}
      osDisk:
        osType: "Linux"
        diskSizeGB: 30
        managedDisk:
          storageAccountType: "Premium_LRS"
      sshPublicKey: ${SSH_PUBLIC_KEY}
      image:
        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
        kind: AzureImageMarketplace
        name: node_image
```
It would be good to discuss your feedback with @devigned, @serbrech and @CecileRobertMichon as the consensus seemed to be separate kinds. Be good to discuss on the call today (i have a meeting clash so may not be able to attend).

Reply:
As long as AzureMarketplaceImage embeds metav1.TypeMeta and is registered with SchemeBuilder.Register, you can use it as raw extension. You can then move fields from AzureMarketplaceImageSpec into AzureMarketplaceImage.

From a usability perspective, it's better to only have to define the AzureMachineTemplate. @devigned @serbrech @CecileRobertMichon what do you think?

Reply:
Thanks @juan-lee, i will modify to use RawExtension and carry on with the implementation.